manifests/deploy/export.pp

```diff
@@ -18,8 +18,8 @@
   -> Openvpn::Client[$name]
   -> Openvpn::Deploy::Export[$name]
 
-  if $facts['openvpn'] {
-    if $facts['openvpn'][$server][$name] {
+  if has_key($facts, 'openvpn') {
+    if dig44($facts, ['openvpn', $server, $name], false) {
```

Comment:
couldn't we just use:
```puppet
if fact("openvpn.${server}.${name}") {
```

Reply:
Would also work out, yeah, dig is just the pattern to use when you can't assert a hash structure in general… of course, `fact` in this specific instance is a more specific way around the whole issue.

(I left the `has_key` bit in as I didn't want to change the error cases as presented so far: fail if the whole fact doesn't exist, otherwise just consider working or not.)